analyses/cell-type-ewings/scripts/clustering-workflow/01-clustering.R

```diff
@@ -0,0 +1,166 @@
+  ), 
+  make_option(
+    opt_str = c("--leiden_cpm"),
+    action = "store_true",
+    default = FALSE,
+    help = "Indicate whether or not to perform Leiden clustering with the CPM objective function."
+  ), 
+  make_option(
+    opt_str = c("--nn_range"),
+    default = "5,10,15,20,25,30,35,40",
+    type = "character", 
+    help = "Comma separated list with a range of values to use for nearest neighbors parameter."
+  ), 
+  make_option(
+    opt_str = c("--louvain_res_range"),
+    default = ".5,1,1.5",
+    type = "character",
+    help = "Comma separated list with a range of values to use for resolution with the Louvain clustering algorithm.
+      Only used with the `--louvain` flag."
+  ),
+  make_option(
+    opt_str = c("--mod_res_range"),
+    default = ".5,1,1.5",
+    type = "character",
+    help = "Comma separated list with a range of values to use for resolution with the Leiden clustering algorithm and modularity objective function.
+      Only used with the `--leiden_mod` flag."
+  ),
+  make_option(
+    opt_str = c("--cpm_res_range"),
+    default = "0.001,0.005,0.01",
+    type = "character",
+    help = "Range of values to use for resolution with the Leiden clustering algorithm and CPM objective function.
+      Only used with the `--louvain` flag."
+  ),
+  make_option(
+    opt_str = c("-t", "--threads"),
+    type = "integer",
+    default = 4,
+    help = "Number of multiprocessing threads to use."
+  ),
+  make_option(
+    opt_str = c("--seed"),
+    type = "integer",
+    default = 2024,
+    help = "A random seed for reproducibility."
+  )
+)
+
+# Parse options
+opt <- parse_args(OptionParser(option_list = option_list))
+
+# Set up -----------------------------------------------------------------------
+
+# set seed
+set.seed(opt$seed)
+
+# make sure inputs exist
+stopifnot(
+  "SCE file does not exist" = file.exists(opt$sce_file),
+  "Must indicate at least one clustering algorithm to use, one of `--louvain`, `--leiden_mod`, or `--leiden_cpm`" = opt$louvain && opt$leiden_cpm && opt$leiden_cpm
+)
+
+# create output directory if it doesn't exist
+output_dir <- dirname(opt$output_file)
+fs::dir_create(output_dir)
+
+
+# read in input sce files
+sce <- readr::read_rds(opt$sce_file)
+
+# get nn and resolution as numbers
+nn_range <- unlist(stringr::str_split(opt$nn_range, ",")) |> 
+  as.numeric()
+
+louvain_res_range <- unlist(stringr::str_split(opt$louvain_res_range, ",")) |> 
+  as.numeric()
+
+mod_res_range <- unlist(stringr::str_split(opt$mod_res_range, ",")) |> 
+  as.numeric()
+
+cpm_res_range <- unlist(stringr::str_split(opt$cpm_res_range, ",")) |> 
+  as.numeric()
+
+# Clustering -------------------------------------------------------------------
+
+# make a list of clustering options
+cluster_opt_list <- list()
+if(opt$louvain){
+  cluster_opt_list$louvain <- list(
+    algorithm = "louvain", 
+    # objective_function = "CPM", # we don't want this to be null 
```

Comment:
I am guessing this should have been uncommented?
```suggestion
    objective_function = "CPM", # we don't want this to be null 
```